src/lib/chat/matrix-client.ts

```diff
@@ -496,14 +499,31 @@ export class MatrixClient implements IChatClient {
     }
   }
 
-  private initializeRoomEventHandlers(room: Room) {
+  private async processRoomTimelineEvent(
+    event: MatrixEvent,
+    _room: Room | undefined,
+    toStartOfTimeline: boolean | undefined,
+    removed: boolean,
+    data: IRoomTimelineData
+  ) {
+    if (removed) return; // only notify for new events, not removed ones
+    if (!data.liveEvent || !!toStartOfTimeline) return; // only notify for new things, not old.
+    if (!this.isSyncing) return; // only notify if syncing is complete
+    if (event.getSender() === this.userId) return; // only notify for other users events
+
+    this.events.receiveLiveRoomEvent(mapToLiveRoomEvent(event) as any);
+  }
+
+  private async initializeRoomEventHandlers(room: Room) {
     if (this.unreadNotificationHandlers[room.roomId]) {
       return;
     }
 
     this.unreadNotificationHandlers[room.roomId] = (unreadNotification) =>
       this.handleUnreadNotifications(room.roomId, unreadNotification);
     room.on(RoomEvent.UnreadNotifications, this.unreadNotificationHandlers[room.roomId]);
+
+    room.on(RoomEvent.Timeline, this.processRoomTimelineEvent.bind(this));
```

Comment:
Do we have to do this in the room specific event registration? Let's move it to the general event registration like the `RoomEvent.Name` event listener.

Reply:
Oh yes of course the `RoomEvent.Timeline` can be moved to the general event registration as its an emitted event, unlike the unread notifications. Updated, thanks.